File: app/src/main/java/at/msd/friehs_bicha/cdcsvparser/App/BaseApp.java
package at.msd.friehs_bicha.cdcsvparser.App;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import at.msd.friehs_bicha.cdcsvparser.transactions.Transaction;
import at.msd.friehs_bicha.cdcsvparser.wallet.CDCWallet;
import at.msd.friehs_bicha.cdcsvparser.wallet.Wallet;

public class BaseApp implements Serializable {

    public ArrayList<Wallet> wallets = new ArrayList<>();
    public ArrayList<Wallet> outsideWallets = new ArrayList<>();
    public ArrayList<Transaction> transactions = new ArrayList<>();
    public int amountTxFailed = 0;

}

File: app/src/main/java/at/msd/friehs_bicha/cdcsvparser/app/BaseApp.kt
package at.msd.friehs_bicha.cdcsvparser.app

import at.msd.friehs_bicha.cdcsvparser.transactions.Transaction
import at.msd.friehs_bicha.cdcsvparser.wallet.Wallet
import java.io.Serializable

/**
 * Base app class
 */
open class BaseApp : Serializable {
    var appType: AppType = AppType.Default
    var id: Int = 0
    open var wallets = ArrayList<Wallet>()  //TODO: remove and replace with Map
    open var walletMap = HashMap<String, Wallet>()
    var outsideWallets = ArrayList<Wallet>()
    open var transactions = ArrayList<Transaction>()
    var failedTxs = ArrayList<String>()
    var amountTxFailed: Long = 0
    var isUseStrictWalletType = false


    /**
     * Adds a failed transaction.
     *
     * @param tx the transaction
     */
    fun addFailedTx(tx: String) {
        failedTxs.add(tx)
        amountTxFailed++
    }

}